auth_backends/helsinki_azure_ad.py

```diff
@@ -1,19 +1,24 @@
+import logging
 from functools import lru_cache
+from json import JSONDecodeError
 
 from django.utils.translation import gettext, pgettext
 from requests import RequestException
-from social_core.backends.azuread_tenant import AzureADTenantOAuth2
+from social_core.backends.azuread_tenant import AzureADV2TenantOAuth2
 from social_core.exceptions import AuthFailed
 
+logger = logging.getLogger(__name__)
 
-class HelsinkiAzureADTenantOAuth2(AzureADTenantOAuth2):
+
+class HelsinkiAzureADTenantOAuth2(AzureADV2TenantOAuth2):
```

Comment:
I suppose the base class change changes the version of the Microsoft OAuth2 API that is used or something like that. This change could perhaps be in its own commit, with an explanation why it's done.

It also seems that the new base class `AzureADV2TenantOAuth2` overrides a couple methods (`get_user_id` and `get_user_details`) from its base class and seems to change behavior somewhat. Is that OK?

Reply:
Separated the change to it's own commit. Yeah the changes are ok.